services/requester/batch_tx_pool.go

```diff
@@ -140,27 +156,50 @@ func (t *BatchTxPool) Add(
 	// For all 3 cases, we record the activity time for the next
 	// transactions that might come from the same EOA.
 	// [X] is equal to the configured `TxBatchInterval` duration.
-	lastActivityTime, found := t.eoaActivity.Get(from)
-
 	if !found {
 		// Case 1. EOA activity not found:
 		err = t.submitSingleTransaction(ctx, hexEncodedTx)
-	} else if time.Since(lastActivityTime) > t.config.TxBatchInterval {
-		// Case 2. EOA activity found AND it was more than [X] seconds ago:
-		err = t.submitSingleTransaction(ctx, hexEncodedTx)
+	} else if time.Since(eoaActivity.lastSubmission) > t.config.TxBatchInterval {
+		// If the EOA has pooled transactions, which are not yet processed,
+		// due to congestion or anything, make sure to include the current
+		// tx on that batch.
+		t.txMux.Lock()
+		hasBatch := len(t.pooledTxs[from]) > 0
+		if hasBatch {
+			userTx := pooledEvmTx{txPayload: hexEncodedTx, nonce: nonce}
+			t.pooledTxs[from] = append(t.pooledTxs[from], userTx)
+		}
+		t.txMux.Unlock()
+
+		// If it wasn't batched, submit individually
+		if !hasBatch {
+			// Case 2. EOA activity found AND it was more than [X] seconds ago:
+			err = t.submitSingleTransaction(ctx, hexEncodedTx)
+		}
 	} else {
 		// Case 3. EOA activity found AND it was less than [X] seconds ago:
-		userTx := pooledEvmTx{txPayload: hexEncodedTx, txHash: tx.Hash(), nonce: tx.Nonce()}
-		// Prevent submission of duplicate transactions, based on their tx hash
-		if slices.Contains(t.pooledTxs[from], userTx) {
-			return errs.ErrDuplicateTransaction
-		}
+		t.txMux.Lock()
+		userTx := pooledEvmTx{txPayload: hexEncodedTx, nonce: nonce}
 		t.pooledTxs[from] = append(t.pooledTxs[from], userTx)
+		t.txMux.Unlock()
+	}
+
+	if err != nil {
+		return err
 	}
 
-	t.eoaActivity.Add(from, time.Now())
+	// Update metadata for the last EOA activity only on successful add/submit.
+	eoaActivity.lastSubmission = time.Now()
+	eoaActivity.txNonces = append(eoaActivity.txNonces, nonce)
+	// To avoid the slice of nonces from growing indefinitely,
+	// maintain only a handful of the last tx nonces.
+	if len(eoaActivity.txNonces) > maxTrackedTxNoncesPerEOA {
+		eoaActivity.txNonces = eoaActivity.txNonces[1:]
+	}
```

Comment:
it _should_ always be 1, but just in case
```suggestion
	// To avoid the slice of nonces from growing indefinitely,
	// keep only the last `maxTrackedTxNoncesPerEOA` nonces.
	if len(eoaActivity.txNonces) > maxTrackedTxNoncesPerEOA {
		firstKeep := len(eoaActivity.txNonces) - maxTrackedTxNoncesPerEOA
		eoaActivity.txNonces = eoaActivity.txNonces[firstKeep:]
	}
```